src/playlist-maker.js

```diff
@@ -133,217 +122,336 @@ const factory = (player, initialList, initialIndex = 0) => {
    * @return {Array}
    *         The playlist
    */
-  const playlist = player.playlist = function(newList, newIndex = 0) {
+  const playlist = player.playlist = (newList, newIndex = 0) => {
     if (Array.isArray(newList)) {
       list = newList.slice();
       if (newIndex !== -1) {
         playlist.currentItem(newIndex);
       }
-      playlist.changeTimeout_ = window.setTimeout(() => {
-        player.trigger('playlistchange');
-      }, 0);
+      player.setTimeout(() => player.trigger('playlistchange'), 1);
```

Comment:
Switched `setTimeout`s to player-level, so we don't need to worry about cleaning them up.